src/cpp/session/SessionMain.cpp

```diff
@@ -1879,6 +1889,15 @@ int main(int argc, char * const argv[])
       ::setpgrp();
 #endif
 
+#ifdef _WIN32
+      // initialize runtime library
+      Version rVersion(getDLLVersion());
+      bool isUcrt = rVersion >= Version("4.2.0");
```

Comment:
Will r >= 4.2 on windows always be ucrt? Aren't there some exceptions based on windows version? I recall Tomas K saying something to that effecr

Reply:
I believe so, but we should probably have a better check for this. The note at https://cran.r-project.org/bin/windows/base/ says:

> This build requires UCRT, which is part of Windows since Windows 10 and Windows Server 2016. On older systems, UCRT has to be installed manually from [here](https://support.microsoft.com/en-us/topic/update-for-universal-c-runtime-in-windows-c0514201-7fe6-95a3-b0a5-287930f3560c).

Let me see if there's a better place for me to figure this out.

Reply:
> I just wanted to clarify that to use
UTF-8 as the native encoding in R 4.2, if you are using long-term
support Windows servers, you need Windows Server 2022 (Windows Server
2016 is not enough). If you run R 4.2 on Windows Server 2016, it will
still use the Windows default encoding (as R 4.1 and older).

> More precisely, as "R for Windows FAQ" says,


> For UTF-8 to be the native encoding, you need R at least 4.2 and at least
Windows 10 (version 1903) on desktop systems, Windows Server 2022 on
long-term support server systems or Windows Server 1903 from the semi-annual
channel.